### ISSUE TITLE
Mostly-Comprehensive List of Bugs / Suggestions (Flans Mod 5.5.2 for Minecraft 1.12.2) 

### ISSUE THREAD

**Issue by cpttuttle** [FlansMods/FlansMod#913]
I have played flans mod for a long time and have enjoyed playing more and more with each successive version. However, I am aware that many glitches and bugs occur when  developing such a complicated mod. Thus, I wanted to make a mostly-comprehensive list of problems I systematically discovered and suggestions I have for the game in order to make game play better in future versions.

I understand how some of these problems are complicated and may require much time to fix and do not expect them to be updated immediately. Other problems may be more simple to change.  In either case, I thought it may be helpful to flag some of these issues so the developer would not have to spend the time sorting through them himself, especially considering how little time he has as expressed on his Patreon page. There are some things I may have missed, but overall I think I have covered most serious issues. _But you can reply if you find something I didn't!_

Note: I downloaded the latest version of the mod (Version 5.5.2 for Minecraft 1.12.2) as of May 16, 2018 from the [flans mod official website](http://flansmod.com/download).

Imperative issues are in **bold**
Items in question are _italicized_
_*_ =  Suggestions

                                                  Weapons
--World War 2--
American
* *_Springfield (Not Sniper)_ -- reduce recoil. The regular Springfield has the same recoil as the sniper which is not the case for other guns like the Lee Enfield or the Kar98K. Also, both the Springfield and the M1 Garand are .30-06, so they should be comparable in recoil. 
* *_Colt_ -- the pistol reloads from the front of the grip rather than the bottom. 
* _M3A1 Grease Gun_ -- the sound is delayed from the shot. 
* _Trenchgun_ -- the trench gun reload sound and visual loading of the shells are mismatched. The player hears 7 shells but 8 are shown to be loaded.

British
* *_Lee Enfield_ -- update model (Side note: I do like the color that simulates the light finish on the real gun).
* *_Webley_ -- the Webley is actually a top break rather than a swing out cylinder revolver. Consider changing the reloading animation. 

Russian
* _TT33_ --update model. It's pretty good but could use some touch up. 
* *_Panzerfaust_ -- is originally a German anti-tank gun and is in the Russian weapon box. I'm not sure if there is a reason for this of which I am not aware. 

German
* _**FG42**_ -- ~model is nearly invisible in both first and third person views~
* _**Luger**_ -- ~model is fully invisible in both first and third person views~
* _**STG 44**_ -- ~model is incomplete/messed up~
* _**Panzerschreck**_ -- ~model is nearly invisible in both first and third person views. Also, it appears to be extremely long.~

Japanese
* *_Type 38_ -- update model. I do like how the sights are pronged; maybe make them more visible when the player left clicks to aim. 
* _Type 100_ -- the shooting sound is choppy. *~Also, update model.~ 
* *_Type 14_ -- update model. It is also strange that the gun reloads the whole grip instead of just a magazine.
* _**Type 99**_ -- ~the model is almost completely white and is identical to the Bren.~
* _Panzerschreck_ (?) -- I'm not sure why the panzerschreck is included in the Japanese box when it is a German rocket launcher. *Maybe consider adding the Type 4 70mm AT Rocket Launcher instead. 

General 
* **_All Mounted Machine Guns_** -- ~they will not shoot where aimed. When mounted, they all point directly at the ground in front of the player and will not shoot where pointed.~ *Also, it'd be neat to show the ammo boxes on the gun when they are loaded. 
* _**All Grenades**_ -- ~model is incomplete/messed up~
* _Flamethrowers_ -- have no sound of their own. *Would be neat to have a "wooshing" flame noise. 

--Modern Warfare--
General
* *_Pistols_ -- update pistol models 
* *LMG's -- when they are shot, where the bullets go verses where the barrel of the gun is pointed are not the same. More specifically, the barrel looks like it is pointing towards the sky in first person mode when the bullets are actually shooting straight.

                                                Armor 
--World War 2 --
* _**General**_ -- ~can not scroll through all of the armor because it crashes Minecraft. Can only view American and British armor. Note: I did not try on every piece of armor to see if they are gltichy or not.~

                                               Vehicles 
--Ground--
* _Dune Buggy_ -- ~the name of the item is "item.buggy.name". Consider renaming.~
* **_M1 Abrams_** -- when placed, the vehicle has no tracks. It only glitches in and out when it is actually driven. 
* **_Leopard 2A6_** -- ~model is incomplete/messed up. A lot. ~
* **_BMWR75_** -- the motorcycle constantly rotates. Also, it does not have any sound.
* _Chi-Ha_ -- the players head sticks out of the roof and can see the sky from inside the cockpit of the tank. 
* _Chi-Nu_ -- the name in the vehicle crafting table is "ChiNu"
* *_Flak88_ -- has no explosion damage on the terrain (and my explosions were on)
* *_KV-1_ -- update model
* _**M1-57 MM**_ -- has no sound when shot
* _M3 Halftrack_ -- item name is "M3Halftrack"
* _**M45 .50Cal Quad Mount**_ -- does not shoot. Probably related to the issue of drivers guns not shooting. 
* _SAS Willy's Jeep_ -- item name is "British Jeep" in the creative inventory. It confused me. And the mounted machine guns on the vehicle shoot the bullets from weird places below the jeep, not actually from the barrel of the gun. 
* _**SdKfz.2**_ -- has no sound when driving
* _**SdKfz.222 Armored Car**_ -- has no sound when driving. Also, it has a giant gun, but no way to load shells or guns.
* _**SU-1-12**_ -- ~model is purple and black.~ Also, the gun does not like to shoot. 
* _**Tiger**_ -- has no sound when shot. Also, does not shoot shells while accelerating / pressing "w"
* _**Tiger 131**_ -- has choppy driving sounds.
* _**Tiger 2 H**_ -- has no sound when shot. Also, does not shoot shells while accelerating / pressing "w"

--Air--
* *_A-10 Warthog_ -- I think it would be neat to add separate key binds to drop bombs from the vehicle and maybe even add stinger missiles since the real A-10 can drop unguided bombs and engage in air-to-air combat with sidewinder missiles
* _**B-52 Bomber**_ -- does not drop any bombs and says there are gun slots when there are no mounted guns to be shot *(Side note: I think there should be more bomb slots since the B-52 bomber can carry 70,000lb (31500kg) of mixed ordnance).
* *_UH-60 Blackhawk_ -- I would suggest adding armaments to the Blackhawk since the real helicopter has up to 6 guns (some for side mounted gunners) and 4 various missiles. 
* *_Chinook_ -- typically have armaments as well. Or at least cargo space. 
* *_Bell AH-1 "Huey" Cobra_ -- update model. Maybe consider adding a UH-1 Iroquois helicopter as well(?)
* *_MH-6 Little Bird_ -- since it is an attack helicopter, I would suggest arming it. Side gunners could be added (?).
* *_BF109_ -- I noticed there is no bomb slot but there is one on the P-51 Mustang. Yet, both were capable of carrying payloads of bombs. 
* _**Lancaster Bomber**_ -- there is no "gun" slot in the interface
* *_Spitfire_ -- update model. You gotta give the British some love too. 

General
* _**Modern Warfare Tanks**_ -- the sound is cut off short when a shell is shot. 
* **_All Vehicles_ -- ~driver's secondary weapons do not work~**

Overall Problems
* ~When windows are shot with bullets, it produces purple and black particles~
* Apocalypse items appear purple and black in player's inventory: Labratory Stone, Sulfur (block), and Power Cube.
* Apocalypse items names are weird: Skeleton, Skeleton2, AI Chip, Complicated Circuit, Nuclear Power Core, and Gun Rack (they appear as "tile.flan(item name).name"
* **Apocalypse survivors do not recognize the player as hostile**
* Entering an airplane that is flying in the apocalypse map causes the game to crash. 
* Consider reorganizing the creative mode items to be more logical. For example, separating it in the order of the weapon boxes or something of that sort. 
* _Duel wield_ is not working for two pistols nor the riot sheld + pistol combination
* _Vehicle spawner_ -- the middle of the vehicles spawn at the spawner, meaning the tires/track/landing gear are in the ground. 
* **_Flag_** -- ~is pink and black~
* ~Weapon modifications in the Gun Modification Table do not like being shift-clicked out of their slots once attached to a gun. It crashes my game.~
* ~Weapons do not appear to be held in hand by the player as they did in previous versions. Put another way, when the player is holding a gun, it will rest by their side in third person view but will appear to be outstretched in first person view.~

**Comment by upupaepops** [comment]
I'm really hope that someday these problems will be fixed...))

**Comment by Gladorme** [comment]
Thanks for listing all problems 😄

**Comment by CaveMiner1215** [comment]
I found another problem.

- All aircraft: Forward Firing armament won't fire, despite being loaded.

**Comment by GTHB317** [comment]
Also for me if a fuel can gets drained in a proto titan, it crashes my game everytime. I have not tried this with anything else yet.

**Comment by ChrisLane** [comment]
Fixed third person view for guns in 30d98fbcfce5623df4508eda899c933ab19fd60b
Weapon mod box shift clicking crash was fixed in 500b1093fe0a2645f40047f2ffda54485341be70
Armour box crash fixed in d02694aa1fe6d254606e82bc2c9735221d05c56f
MG resting/shooting angles fixed in bd9905a3893f2671516f621d7d6de63642be5914

**Comment by ChrisLane** [comment]
Fixed a lot of the texture/model issues with guns and vehicles.

**Comment by euanPC** [comment]
in the 1.12 version of the modern warfare pack the automated minigun turrets spawn 2 blocks above where  
they were placed and do not fire

**Comment by ChrisLane** [comment]
@euanPC That's fixed. See #983

**Comment by euanPC** [comment]
I think some of the crossed out problems still exist in the 1.12 releases such as the gun mounts facing down

**Comment by ChrisLane** [comment]
That's definitely fixed. You need the beta version from the releases page on GitHub

**Comment by euanPC** [comment]
@ChrisLane
how do I download the beta version?

**Comment by ChrisLane** [comment]
> @ChrisLane
> how do I download the beta version?

https://github.com/FlansMods/FlansMod/releases

**Comment by ghost** [comment]
The issue with drivers secondary weapons is still around and it is extremely frustrating.
E.g of it, main armaments work (missiles, tank cannons etc) but guns like aircraft's main guns (e.g minigun on a-10, drivers guns on ww2 aircraft) DO NOT WORK.  Also, guns that are controlled by another player for example the minigun on the huey cobra work FINE.

**Comment by ChrisLane** [comment]
@fancyabum As far as I can see, I just fixed vehicle weapons.

**Comment by Dutchkills** [comment]
hi chrislane i downloaded the 5.6 flans mod but when i start the game i get this error

net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Flan's Mod (flansmod)

now i loaded the new packs with flans version 5.5.2 and then it works

the one who did the upgraded german gun models did a GREAT JOB

**Comment by ChrisLane** [comment]
@Dutchkills please provide a crash log in a new issue.

**Comment by Dutchkills** [comment]
i made a new issue

**Comment by Qile0317** [comment]
I'm not sure where to post this, but does anyone know if either new vehicles are going to be added in the "default" content packs or if other content pack creators are going to update the packs to 1.12.2 as well?

**Comment by ghost** [comment]
I'm still getting purple/black checkerboard pattern on the SU-1-12 with the latest update.

**Comment by Gonino** [comment]
In the 5.5.2 i have the same problem of @Jprik  (SU-1-12 purple/black) and the Flag is a Purple flag with Purple/Black stick. I can't use the Soviet gun DP-28 and i still having the @fancyabum problem (Airplane guns don't work)

Can i play with the 5.56 pre-release version, or is only for coding?

Thank you

**Comment by Liruxo** [comment]
Version 5.5.2 is out of date and no longer supported. Please download the newest version of FlansMod here: https://github.com/FlansMods/FlansMod/releases

For the "Airplanes Guns dont work"- Issue please refer to #1037

**Comment by Gonino** [comment]
Thank you! Regarding that, I would like to ask you something

I don't know how to install the latest version 5.6. I downloaded the folder that you linked and i converted the Packs in "run" folder, to .jar and replace the others in my game. But that doesn't work and now i haven't any items of Flans mod when i play the game. (i guess the other attached files are for coding)

**Comment by MrGlayden** [comment]
Has anyone managed to get a working version?  
Im currently using 5.5.2 but I tried a few of the others later releases where it says "fixed vehicles not firing" only to find they are in some cases in a worse state then my broken version.  
Tanks/helis/planes will all *technically* fire their missiles and bombs, but they explode immediately after being fired, my first though was that they were colliding with the vehicle they are fired from, but im not a coder so I dont really know

**Comment by Liruxo** [comment]
@MrGlayden These bugs are fixed in release v5.6-beta.6. You can download this Pre-Release at https://github.com/FlansMods/FlansMod/releases

**Comment by MrGlayden** [comment]
I had tried downloading the newest release and i wasnt even able to fire in that version, tried a few earlier releases but later then the 1 i was using and had similar issues, ill try again when i get home though and ill record video if still not working my end

**Comment by MrGlayden** [comment]
@Liruxo  Ok so I tried with that latest release and the weapons *did* fire, but the game crashed everytime they hit something with a "ticking error"  
  
The crash report pretty much is just "at com.flansmod.common.guns.raytracing.FlansModRaytracer.raytraceBlock(FlansModRaytracer.java:207) " about 500 times.  
  
Is there any known conflict issues with other mods that could be causing my issue here?

**Comment by ChrisLane** [comment]
> @Liruxo Ok so I tried with that latest release and the weapons _did_ fire, but the game crashed everytime they hit something with a "ticking error"
> 
> The crash report pretty much is just "at com.flansmod.common.guns.raytracing.FlansModRaytracer.raytraceBlock(FlansModRaytracer.java:207) " about 500 times.
> 
> Is there any known conflict issues with other mods that could be causing my issue here?

Please remove all other mods to see if that solved the issue, if it does then please use process of elimination to find out which.
Please open an issue with the crash report template if one does not exist already and provide the full crash report.

**Comment by ChrisLane** [comment]
Closing this issue since it's too hard to follow.
For issues that still remain and don't already have an issue opened for them, please open one.

**Comment by MrGlayden** [comment]
> 
> 
> > @Liruxo Ok so I tried with that latest release and the weapons _did_ fire, but the game crashed everytime they hit something with a "ticking error"
> > The crash report pretty much is just "at com.flansmod.common.guns.raytracing.FlansModRaytracer.raytraceBlock(FlansModRaytracer.java:207) " about 500 times.
> > Is there any known conflict issues with other mods that could be causing my issue here?
> 
> Please remove all other mods to see if that solved the issue, if it does then please use process of elimination to find out which.
> Please open an issue with the crash report template if one does not exist already and provide the full crash report.

---------------------------------------------------------

Found the issue, 
malisiscore-1.12.2-6.5.1
malisisdoors-1.12.2-7.3.0  
With these removed it works again

**Comment by ghost** [comment]
I have an error with machine guns, because I can't shoot up  them help me?

**Comment by ghost** [comment]
Flan=WW2_Content_Pack_1.12.2_5.5.2

**Comment by ghost** [comment]
Flan's Mod-1.12.2-5.5.2

**Comment by ghost** [comment]
forge 1.12.2-14.23.5.2847

**Comment by Liruxo** [comment]
Please do not write multiple messages, use the "Edit"-Function istead. Please do not use closed (!) bug reports to get help, instead open a new Bug Report.

Concerning your issue it seems you are using an old & outdated version of FlansMod. Please head over to https://www.github.com/FlansMods/FlansMod/releases and download the newest version.

Please do not answer if you still have issues, instead open a new Bug Report.

**Comment by euanPC** [comment]
is 1.7.10 no longer updated

**Comment by ChrisLane** [comment]
> is 1.7.10 no longer updated

Yes

**Comment by euanPC** [comment]
does the modern warfare server no longer work because it's on 1.15 with a whitelist

**Comment by euanPC** [comment]
does the modern warfare server no longer work because it's on 1.15 with a whitelist

**Comment by Z-D4WG** [comment]
My flans mod only shows colored wool and my content packs did download successively but won't work on minecraft.  can someone help?  the content packs i installed are the ww11 simple parts and nerf.

**Comment by Liruxo** [comment]
Please check against these instructions https://github.com/FlansMods/FlansMod/wiki/Installing-FlansMod
If the issue persists, please follow these steps: https://github.com/FlansMods/FlansMod/wiki/Issue-Reporting

**Comment by Myszakisame** [comment]
I have fixed the Lancaster gun turrets not working. It is actually very easy to do, so if you wanna use this oldschool mod, i suggest you do the following:

1. Open the WW2 Content pack in Flan folder with WinRaR or some similiar tool.
2. Go to the Planes folder and open Lancaster.txt (make a backup before editing! Also if you play on any server with the official release either change the txt on the server side too or use the backup as... well, backup.)
3. Go to the ----------------Passengers------------------- section
4. Edit Passenger 1, Passenger 4 and Passenger 5 - Replace _browning30Cal_ with _browning_.
5. Save, profit.
Found that when i saw that the spitfire also used the 30 cal and it worked. Spitfire referred to it as just browning. It is such an easy fix that it blows my head how that was not included in the release. Took me 10 minutes to figure that out. You can also change for example the BF109 not having any bombs using the same method, it's under the "inventory" section
